### DISCUSSION TITLE
Covariate GPLVM

### DISCUSSION THREAD

**Discussion by Acolome** [GPflow/GPflow#1818] [Q&A] [answered]
Hi,

I am trying to implement a GPflow version of the paper "Decomposing feature-level variation with Covariate Gaussian Process Latent Variable Models", by K. Marteens et al. 

In short, they use GPLVM but with context features added to the kernel variables. This results in a latent space dimension of D (the latent space itself) + the dimension of the context features. 

Therefore, either using GPLVM of BayesianGPLVM, I need to set trainable = false to part of the latent variables. 

As an example, when building a GPLVM:
gplvm = gpflow.models.GPLVM(
    DATA, 
    latent_dim=latent_dim+D_context,
    kernel=k # this kernel already includes both the features and latent variable and active dims according to each kernel
) 

One of the trainable variables shown with print_summary is GPLVM.data[0] . But this includes both latent variables and features in a tensor shape of dimension N x (latent_dim+D_context), of which I only want to train the part corresponding to the latent dims.

Is there a way I can tell the gplvm model to only train part of that tensor? With BayesianGPLVM it's basically the same problem.

Thanks.

A.

**Comment by st--** [comment, marked as answer]
You could have a Variable for the trainable part and a fixed tensor for the other, and then use TensorFlow's concat function to combine the two into the full array you need.
